src/mozilla_sec_eia/models/sec_eia_record_linkage/transform_sec_input.py

```diff
@@ -0,0 +1,392 @@
+    ex21_df = flatten_companies_across_time(
+        df=ex21_df, key_cols=["sec_company_id"], date_col="report_year"
+    )
+    ex21_df = ex21_df.fillna(np.nan)
+
+    return ex21_df
+
+
+def transform_basic10k_table(
+    basic_10k_df: pd.DataFrame, sec10k_filing_metadata: pd.DataFrame
+) -> pd.DataFrame:
+    """Transformations on SEC basic 10K filer table to prepare for record linkage."""
+    basic_10k_df = basic_10k_df.reset_index().pivot_table(
+        values="value", index="filename", columns="key", aggfunc="first"
+    )
+    basic_10k_df.columns.name = None
+    basic_10k_df = (
+        basic_10k_df.reset_index()
+        .pipe(_remove_weird_sec_cols)
+        .pipe(_add_report_year_to_sec, sec10k_filing_metadata)
+        .rename(columns=SEC_COL_MAP)
+        .pipe(clean_location_of_inc)
+        .pipe(transform_company_name)
+        .assign(
+            zip_code=lambda df: df["zip_code"].str[:5],
+            files_10k=True,
+            sec_company_id=lambda df: df["central_index_key"],
+        )
+        .pipe(fill_street_address_nulls)
+    )
+    basic_10k_df[STR_COLS] = basic_10k_df[STR_COLS].apply(
+        lambda x: x.str.strip().str.lower()
+    )
+    basic_10k_df["street_address"] = expand_street_name_abbreviations(
+        basic_10k_df["street_address"]
+    )
+    # flatten across time on unique company name and address pair
+    basic_10k_df = flatten_companies_across_time(
+        df=basic_10k_df, key_cols=["company_name", "street_address"]
+    )
+
+    return basic_10k_df
+
+
+@asset(
+    ins={
+        "basic_10k_dfs": AssetIn("basic_10k_company_info"),
+        "sec10k_filing_metadata_dfs": AssetIn("sec10k_filing_metadata"),
+    },
+)
+def core_sec_10k__filers(
+    basic_10k_dfs: dict[str, pd.DataFrame],
+    sec10k_filing_metadata_dfs: dict[str, pd.DataFrame],
+) -> pd.DataFrame:
+    """Asset for creating a cleaned basic 10k table with EIA utility matched.
+
+    Flatten the table across time to only keep the most recent record
+    for each unique company name and address pair. Clean table and link filers
+    to EIA utilities.
+    """
+    basic_10k_df = pd.concat(basic_10k_dfs.values())
+    sec10k_filing_metadata = pd.concat(sec10k_filing_metadata_dfs.values())
+    basic_10k_df = transform_basic10k_table(basic_10k_df, sec10k_filing_metadata)
+    out_df = basic_10k_df.fillna(np.nan).reset_index(names="record_id")
+    # match EIA utilities to filers
+    # TODO: Here we conduct the match to EIA and add on a column with utility_id_eia
+    return out_df
+
+
+@asset(
+    ins={
+        "sec_10k_filers_matched_df": AssetIn("core_sec_10k__filers"),
+        "clean_ex21_df": AssetIn("transformed_ex21_subsidiary_table"),
+        "clean_eia_df": AssetIn("core_eia__parents_and_subsidiaries"),
+    },
+)
+def out_sec_10k__parents_and_subsidiaries(
+    sec_10k_filers_matched_df: pd.DataFrame,
+    clean_ex21_df: pd.DataFrame,
+    clean_eia_df: pd.DataFrame,
+) -> pd.DataFrame:
```

Comment:
Maybe this final output asset should live in a different module and not be grouped in with the transformations for the SEC input?

Reply:
I think it's probably fine as is. There's probably broader reorganization stuff we could be doing, but I'm not sure that we have time for it right now.